src/services/document_service.py

```diff
@@ -370,7 +370,14 @@ def remove_by_path(self, data_source_id: str, directory: str):
             parent_uid = get_document_uid_by_path(f"{'/'.join(directory.split('/')[0:-1])}", data_source)
             child_uid = get_document_uid_by_path(directory, data_source)
             parent_node = self.get_node_by_uid(data_source_id, parent_uid)
-            parent_node.children[0].remove_by_child_id(child_uid)  # The first child of a directory is always 'content'
+            child_node_id = None
+
+            for child in parent_node.children[0].children:
+                if child.uid == child_uid:
+                    child_node_id = child.node_id
```

Comment:
For performance and readability, these kind of "find first match" should be written with the "next()" function